audiomentations/augmentations/air_absorption.py

```diff
@@ -0,0 +1,177 @@
+        max_temperature: float = 20.0,
+        min_humidity: float = 30.0,
+        max_humidity: float = 90.0,
+        min_distance: float = 10.0,
+        max_distance: float = 100.0,
+        p=0.5,
+    ):
+        """
+        :param min_temperature: Minimum temperature in celcius (can take a value of either 10 or 20)
+        :param max_temperature: Maximum temperature in celsius (can take a value of either 10 or 20)
+        :param min_humidity: Minimum humidity in percent (between 30 and 90)
+        :param max_humidity: Maximum humidity in percent (between 30 and 90)
+        :param min_distance: Minimum microphone-source distance in meters.
+        :param max_distance: Maximum microphone-source distance in meters.
+        :param p: The probability of applying this transform
+        """
+        assert float(min_temperature) in [
+            10.0,
+            20.0,
+        ], "Sorry, the only supported temperatures are either 10 or 20 degrees celcius"
+        assert float(max_temperature) in [
+            10.0,
+            20.0,
+        ], "Sorry, the only supported temperatures are either 10 or 20 degrees celcius"
+        assert min_temperature <= max_temperature
+        assert 30 <= min_humidity <= max_humidity <= 90
+
+        super().__init__(p)
+
+        self.min_temperature = min_temperature
+        self.max_temperature = max_temperature
+
+        self.min_humidity = min_humidity
+        self.max_humidity = max_humidity
+
+        self.min_distance = min_distance
+        self.max_distance = max_distance
+
+    def randomize_parameters(self, samples: np.ndarray, sample_rate: int) -> np.ndarray:
+        super().randomize_parameters(samples, sample_rate)
+        self.parameters["temperature"] = 10 * np.random.randint(
+            int(self.min_temperature) // 10, int(self.max_temperature) // 10 + 1
+        )
+        self.parameters["humidity"] = np.random.randint(
+            self.min_humidity, self.max_humidity + 1
+        )
+        self.parameters["distance"] = np.random.uniform(
+            self.min_distance, self.max_distance
+        )
+
+    def apply(self, samples: np.ndarray, sample_rate: int):
+        assert samples.dtype == np.float32
+
+        humidity = self.parameters["humidity"]
+        distance = self.parameters["distance"]
+
+        # Choose correct absorption coefficients
+        key = str(int(self.parameters["temperature"])) + "C"
+        bounds = [30, 50, 70, 90]
+        for n in range(1, len(bounds)):
+            if bounds[n - 1] <= humidity or humidity <= bounds[n]:
+                key += f"_{bounds[n-1]}-{bounds[n]}%"
+                break
+
+        # Convert to attenuations
+        attenuation_values = np.exp(
+            -distance * np.array(self.air_absorption_table[key])
+        )
+
+        # Calculate n_fft so that the lowest band can be stored in a single
+        # fft bin.
+        first_band_bw = self.air_absorption_table["center_freqs"][0] * 2
```

Comment:
If your first center frequency is 125 Hz, then your octave band bandwidth is 88 - 176 Hz, which is not 125*2. Or maybe `first_band_bw `refers to something else ?

Reply:
You are right. Nice catch.